qiskit_dynamics/pulse_simulator/pulse_simulator.py

```diff
@@ -0,0 +1,526 @@
+        Returns:
+            DynamicsJob object containing results and status.
+        Raises:
+            QiskitError: If invalid options are set.
+        """
+
+        if validate:
+            _validate_run_input(run_input)
+
+        # Configure run options for simulation
+        if options:
+            backend = copy.copy(self)
+            backend.set_options(**options)
+        else:
+            backend = self
+
+        schedules, num_memory_slots_list = _to_schedule_list(run_input, backend=backend)
+
+        # get the acquires instructions and simulation times
+        t_span, measurement_subsystems_list, memory_slot_indices_list = _get_acquire_data(
+            schedules, backend.options.subsystem_labels
+        )
+
+        # Build and submit job
+        job_id = str(uuid.uuid4())
+        dynamics_job = DynamicsJob(
+            backend=backend,
+            job_id=job_id,
+            fn=backend._run,
+            fn_kwargs={
+                "t_span": t_span,
+                "schedules": schedules,
+                "measurement_subsystems_list": measurement_subsystems_list,
+                "memory_slot_indices_list": memory_slot_indices_list,
+                "num_memory_slots_list": num_memory_slots_list,
+            },
+        )
+        dynamics_job.submit()
+
+        return dynamics_job
+
+    def _run(
+        self,
+        job_id,
+        t_span,
+        schedules,
+        measurement_subsystems_list,
+        memory_slot_indices_list,
+        num_memory_slots_list,
+    ) -> Result:
+        """Simulate a list of schedules."""
+
+        # simulate all schedules
+        y0 = self.options.initial_state
+        if y0 == "ground_state":
+            y0 = Statevector(self._dressed_states[:, 0])
+
+        solver_results = self.options.solver.solve(
+            t_span=t_span, y0=y0, signals=schedules, **self.options.solver_options
+        )
+
+        # compute results for each experiment
+        experiment_names = [schedule.name for schedule in schedules]
+        rng = np.random.default_rng(self.options.seed_simulator)
+        experiment_results = []
+        for (
+            experiment_name,
+            solver_result,
+            measurement_subsystems,
+            memory_slot_indices,
+            num_memory_slots,
+        ) in zip(
+            experiment_names,
+            solver_results,
+            measurement_subsystems_list,
+            memory_slot_indices_list,
+            num_memory_slots_list,
+        ):
+            experiment_results.append(
+                self.options.experiment_result_function(
+                    experiment_name,
+                    solver_result,
+                    measurement_subsystems,
+                    memory_slot_indices,
+                    num_memory_slots,
+                    self,
+                    seed=rng.integers(low=0, high=9223372036854775807),
```

Comment:
Im not entirely sure whats going on here, but wanted to check that you still get deterministic measurement outcome if you set the `seed_simulator` option?

Reply:
Yep. The `rng` is instantiated above using the `seed_simulator` option. This line then generates another valid seed using `rng` (a 64 bit positive integer, hence the weird high number) to pass to the results function to use for generating counts for the current experiment in the loop.

Reply:
The tests already use this - they check exact dictionary equality, which only works due to the determinism from the seeds.